### PR TITLE
[WIP] Fixes RCD giving no user feedback when out of ammo

### PR DESCRIPTION
**Pull request by AzuleUtama**
**What does this PR do:**
Fix for issue #10021

RCD will now give a low ammo message for what you are trying to build / destroy. Additionally gives messages when trying to build airlocks or windows over space tiles otherwise it would give a low ammo warning when trying to do so when you actually had enough matter in the RCD.

This did require working the original code for building airlocks slightly, originally the check for enough matter and whether or not there was a floor were contained in the same if statement. Now it first checks for the floor and then calls a second if statement for the matter. This is done so it can have separate errors for trying to build without ammo and without a floor underneath.

I tested this out with many airlock variants and trying to build every combination I could with the tool. I also used the RCD variants including the one borgs get, which gave the same messages when the borg was out of power. I didn't experience any errors when doing so and all the new messages displayed as expected.

Please let me know if this all looks okay or if there's anything glaringly wrong with it that I missed.

**Images of sprite/map changes (IF APPLICABLE):**

No changes to sprites.

**Changelog:**
:cl::
fix: RCD now gives visible messages when building without enough ammo or on an invalid tile.
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/RCD.dm [hunk 1]
```diff
@@ -392,6 +419,10 @@ GLOBAL_LIST_INIT(rcd_door_types, list(
 		var/turf/AT = A
 		AT.ChangeTurf(/turf/simulated/floor/plating) // Platings go under windows.
 		return 1
+	else
```
Comment (tigercat2000) [inline code/game/objects/items/weapons/RCD.dm]: All of these `else`'s you've added are pointless. Just have the to_chat and play_sound where the `return 0` was before. The code was designed to work well in a early return system already.

Comment (AzuleUtama) [reply code/game/objects/items/weapons/RCD.dm]: Sorry, didn't realise they weren't necessary. I'll go ahead and update the code to remove all the `elses` I added in and instead have it go straight to the `to_chat` and `play_sound` section before returning `false` or `0`.

I'll test the whole thing again afterwoods before pushing the updated code here.

Comment (tigercat2000) [reply code/game/objects/items/weapons/RCD.dm]: 👍

Comment (tigercat2000) [reply code/game/objects/items/weapons/RCD.dm]: Your second example is good.

**Comment by AzuleUtama** [comment]
Code updated, now with 100% less else statements!

Some minor points:

- It turns out that the invalid floor message actually displays when you build on any kind of tile the RCD doesn't like, such as trying to build an airlock inside a wall. The error message it gives out has been altered to better reflect this.

- Due to above, a new check was added if you try building a wall somewhere it can't be built.

Once again, if something here looks horribly wrong, please let me know.

**Pull request merged**